### ISSUE TITLE
[Help wanted] Spring Boot - TestContainers - Simple BearerToken?

### ISSUE THREAD

**Issue by Stromner** [navikt/mock-oauth2-server#573]
This is probably a super simple issue but I just don't understand the documentation that exists on mock-ouath2-server, really need help...

What I want to do is just to set up a very simple security validation for my Spring Boot application, to begin with. Where no user detail would need to be provided by the client, just a request to the auth server from the client to get a valid bearer token that can pass the validation step that Spring Boot requires.

**Server**
application.yaml:
```
spring:
  security:
    oauth2:
      resourceserver:
        jwt:
          issuer-set-uri: http://localhost:8080/default
          jwk-set-uri: http://localhost:8080/default/jwks
```
As I figured I don't need any customization as of yet I'm using the default beans for security so my security config is just a shell for now:
```
@Configuration
@EnableWebFluxSecurity
public class SecurityConfig {}
```
**Test module**
My TestContainer that wraps the mock-ouath2-server image looks like this (belongs to same network as the Application container):
```
public class AuthorizationContainer extends GenericContainer<AuthorizationContainer> {
    private static final DockerImageName AUTHORIZATION_IMAGE = DockerImageName.parse("ghcr.io/navikt/mock-oauth2-server:2.0.0");
    private static final int APP_PORT = 8080;
    public AuthorizationContainer() {
        super(AUTHORIZATION_IMAGE );

        addFixedExposedPort(APP_PORT, APP_PORT);
        waitingFor(Wait.forHttp("/isalive").forStatusCode(200));
    }
}
```
**How do you create a valid bearer token with this setup?**

**Comment by ybelMekk** [comment]
Not sure i understand this correctly, but i think @tommytroen can give it an 👁️ ?

**Comment by imsherlocked** [comment]
Hello @Stromner  can you please assign this issue to me, I would deliver it at the earliest.

**Comment by cjenkscybercom** [comment]
Just wanted to point out that I did this with a beforeEach where I used the whatever client I had (I want to say it was RestTemplate, doesn't really matter) and just called the token endpoint like I would in Postman to get a token before tests. 

But I also needed to set the issuer claim (iss) directly because of how the networking works in TestContainers/Docker. I wasl also doing other stuff as well, so I also had the raw JSON string as text in the test file so I could update it before each test as needed.

Was a bit janky but it did work in testing fully end-to-end with bearer tokens.
